plugins/outputs/elasticsearch/elasticsearch.go

```diff
@@ -411,6 +450,13 @@ func (a *Elasticsearch) GetIndexName(indexName string, eventTime time.Time, tagK
 		indexName = dateReplacer.Replace(indexName)
 	}
 
+	if strings.Contains(indexName, "metric_name") {
+		var metricNameReplacer = strings.NewReplacer(
+			"metric_name", metricName,
+		)
+		indexName = metricNameReplacer.Replace(indexName)
+	}
+
```

Comment:
This needs to be documented in the README.md. I see you mentioned it in the PR description but that won't be available to the end user.